torch/_subclasses/fake_tensor.py

```diff
@@ -366,7 +380,7 @@ def run_and_return_new_tensor_of_input_device(fake_mode, func, args, kwargs):
     )
 
     out_device = new_kwargs["input"].device
-    with in_kernel_invocation_manager(fake_mode):
+    with no_dispatch(), in_kernel_invocation_manager(fake_mode):
```

Comment:
So is the invariant now that you always `no_dispatch` whenever you `in_kernel_invocation_manager`? Can we combine the context managers together then?